src/app/blog/BlogClient.tsx

```diff
@@ -52,42 +77,72 @@ export function BlogClient({ posts }: { posts: BlogPostData[] }) {
         type="search"
         id="search"
         name="search"
+        value={searchQuery}
         aria-label="Search blog posts"
         className="text-brand-800"
         onChange={handleSearch}
       />
-      <ul>
-        {filteredPosts.slice(0, visibleCount).map((post) => (
-          <li key={post.slug}>
-            {post.image.url && (
-              <Image
-                src={post.image.url}
-                alt={post.image.alt}
-                width={282}
-                height={141}
-                className="object-cover"
-              />
-            )}
-            <Heading tag="h3" variant="lg">
-              {post.title}
-            </Heading>
-            <p>{post.description}</p>
-            {post.publishedOn && (
-              <span className="block">
-                {formatDate(post.publishedOn, 'blog')}
-              </span>
-            )}
-            <TextLink href={`${PATHS.BLOG.path}/${post.slug}`}>
-              Read More
-            </TextLink>
-          </li>
-        ))}
-      </ul>
-      {hasMorePosts && (
-        <Button aria-label="Load more posts" onClick={handleLoadMore}>
-          Load More
-        </Button>
+
+      <br />
+      <br />
+
+      <CardLayout type="blogPost">
+        {filteredPosts.map((post, i) => {
+          const { image, title, description, slug, publishedOn } = post
+
+          return (
+            <li
+              key={post.slug}
+              className={clsx(
+                'h-[400px] overflow-clip rounded-md border border-brand-600 p-4',
+                // Only show 20 posts at a time, the rest are hidden with sr-only
+                i >= (currentPage - 1) * POSTS_PER_LOAD &&
+                  i < currentPage * POSTS_PER_LOAD
+                  ? 'block'
+                  : 'sr-only',
+              )}
```

Comment:
If we're leaving a comment, it's a good signal that this probably should be abstracted into a more clearly defined function.

Reply:
Yes, good point!